synapse/lib/config.py

```diff
@@ -1,8 +1,126 @@
 import os
+import copy
+import typing
+import logging
+
+import yaml
 
 import synapse.exc as s_exc
 import synapse.common as s_common
 
+logger = logging.getLogger(__name__)
+
+class ConfTypes(object):
```

Comment:
If we're going to move toward configs having better structure, this might be the time to integrate json schema checking type stuff.  My initial poking around found it to be very powerful.